tests/auth_tests/test_checks.py

```diff
@@ -118,6 +119,16 @@ def is_authenticated(self):
             ),
         ])
 
+    @override_settings(AUTH_USER_MODEL='auth_tests.UniqueConstraintUser')
+    def test_user_non_unique_with_model_constraint(self):
+        class UniqueConstraintUser(AbstractBaseUser):
+            username = models.CharField(max_length=30)
+            USERNAME_FIELD = 'username'
+
+            class Meta:
+                constraints = [UniqueConstraint(fields=['username'], name='user_username_unq')]
+        self.assertFalse(checks.run_checks(app_configs=self.apps.get_app_configs()))
```

Comment:
`self.assertFalse()` -> `self.assertIs(..., False)`

Reply:
`run_checks` actually returns an empty list in this case, I changed it to `assertListEqual(errors, [])`.